api/admin/controller/collection_settings.py

```diff
@@ -0,0 +1,266 @@
+            self._db.rollback()
+            return PROTOCOL_DOES_NOT_SUPPORT_PARENTS
+
+        if parent_id:
+            parent = get_one(self._db, Collection, id=parent_id)
+            if not parent:
+                self._db.rollback()
+                return MISSING_PARENT
+            collection.parent = parent
+            settings = protocol.get("child_settings")
+        else:
+            collection.parent = None
+            settings = protocol.get("settings")
+
+        return settings
+
+    def validate_external_account_id_setting(self, value, setting):
+        if not value and not setting.get("optional"):
+            # Roll back any changes to the collection that have already been made.
+            self._db.rollback()
+            return INCOMPLETE_CONFIGURATION.detailed(
+                _("The collection configuration is missing a required setting: %(setting)s",
+                  setting=setting.get("label")))
+
+    def get_integration_id(self, value):
+        if value == self.NO_MIRROR_INTEGRATION:
+            integration_id = None
+        else:
+            integration = get_one(
+                self._db, ExternalIntegration, id=value
+            )
+            if not integration:
+                self._db.rollback()
+                return MISSING_SERVICE
+            if integration.goal != ExternalIntegration.STORAGE_GOAL:
+                self._db.rollback()
+                return INTEGRATION_GOAL_CONFLICT
+            return integration.id
+
+    def process_settings(self, settings, collection):
+        for setting in settings:
+            key = setting.get("key")
+            value = flask.request.form.get(key)
+            if key == "external_account_id":
+                error = self.validate_external_account_id_setting(value, setting)
+                if error:
+                    return error
+                collection.external_account_id = value
+            elif key == 'mirror_integration_id':
+                integration_id = self.get_integration_id(value)
+                if isinstance(integration_id, ProblemDetail):
+                    return integration_id
+                collection.mirror_integration_id = integration_id
+            else:
+                result = self._set_integration_setting(collection.external_integration, setting)
+                if isinstance(result, ProblemDetail):
+                    return result
+
+    def process_libraries(self, protocol, collection):
+        libraries = []
+        if flask.request.form.get("libraries"):
+            libraries = json.loads(flask.request.form.get("libraries"))
+
+        for library_info in libraries:
+            library = get_one(self._db, Library, short_name=library_info.get("short_name"))
+            if not library:
+                return NO_SUCH_LIBRARY.detailed(_("You attempted to add the collection to %(library_short_name)s, but it does not exist.", library_short_name=library_info.get("short_name")))
```

Comment:
This isn't your code, but change this to make it clear that the "it" that doesn't exist is the library, not the collection.